src/func.rs

```diff
@@ -81,6 +86,78 @@ impl Func {
         }
     }
 
+    /// Calls the function using native rust types.
+    pub fn call_typed<C, Params, Ret>(&self, mut ctx: C, args: Params) -> Result<Ret>
```

Comment:
Instead of introducing this new function, would it make more sense in the long-run to replace `TypedFunc::call` with this implementation? Otherwise, this draft looks promising so far!

Reply:
Yes absolutely, it would allow all the same functionally, as well as using strong rust structs and tuples, or even structs of values, which will make everything easier since you can use a common struct for what you know the type of, and then a dynamically typed field in the struct or tuple

Reply:
I've kept the current one as is to keep things working while I move things bit by bit.